cpp/BUILD_MANUAL.md

```diff
@@ -0,0 +1,119 @@
+# CPP Build Manual
+
+The purpose of  this manual is to provide you with some information on how to compile, build, and run the C++ version of the Kata. 
+
+## Prerequisites 
+
+### Build Tools 
+
+[CMake](https://cmake.org/) is our main build tool for the C++ version of this Kata. 
+In addition to CMake, we are using [vcpkg](https://github.com/microsoft/vcpkg) to avoid compiling our third-party libraries.  
+
+Don't worry if you don't have those tools installed! We provided you with a setup script that handles compiling and building the project. 
+ 
+### Required Versions 
+
+#### C++ 
+
+By default, our CMake file is referencing C++ version 17. But, we also tested and 
+compiled the code with versions 11 and 14. 
+
+To use a version other than 17, simply modify the below line in the [CMake](./CMakeLists.txt)
+file to reference 11 or 14.  
+
+```shell
+set(CMAKE_CXX_STANDARD 17)
+```
+
+#### CMake
+	1. Minimum required: 3.10.0
+	1. Used: 3.18.4
+
+### IDEs 
+We have successfully tested loading, compiling and running this kata on the below 3 IDEs: 
+1. [Microsoft Visual Studio](https://visualstudio.microsoft.com/)
+1. [Xcode](https://developer.apple.com/xcode/)
+1. [Visual Studio Code](https://code.visualstudio.com/)
+
+***Note:*** 
+
+1. When executed on Windows and Mac, our script will, by default, generate the solution files ***Mikado-TestBuilders-Workshop.sln*** for **'Visual Studio 2017'** & ***Mikado-TestBuilders-Workshop.xxcodeproj*** for **'Xcode'** respectively. 
+1. The script doesn't generate the VSCode Workspace files. However, loading and running the project into VSCode is straight forward. You can check the tutorial '[Getting started with CMake Tools on Linux](https://code.visualstudio.com/docs/cpp/cmake-linux#:~:text=ready%20to%20build.-,Open%20the%20Command%20Palette%20(Ctrl%2BShift%2BP)%20and,CMake%20Tools%20builds%20all%20targets.)' for reference. 
+
+## Process 
+
+> ***Reminder***:  You need to run the commands below from the [cpp](.) folder!
```

Comment:
Actually this is no longer needed for the cpp_setup.sh. It can be run from any directory in the repo. But anyway it does not hurt to keep this reminder here.